core/input/input.cpp

```diff
@@ -140,7 +331,10 @@ void Input::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("parse_input_event", "event"), &Input::parse_input_event);
 	ClassDB::bind_method(D_METHOD("set_use_accumulated_input", "enable"), &Input::set_use_accumulated_input);
 	ClassDB::bind_method(D_METHOD("is_using_accumulated_input"), &Input::is_using_accumulated_input);
-	ClassDB::bind_method(D_METHOD("flush_buffered_events"), &Input::flush_buffered_events);
+
+	// For backward compatibility this calls the force method.
+	// Use this at your peril as it will mess up agile input any frame it is called.
+	ClassDB::bind_method(D_METHOD("flush_buffered_events"), &Input::force_flush_buffered_events);
```

Comment:
Should we add a deprecated warning, and provide an alternative for users?

Reply:
It's tough to know what to do here with the situation of users creating their own main loop, as the existing API assumes no agile input.

If they are writing a mainloop from a bound language, I'm assuming they currently just call `flush_buffered_events()`, which in turn calls `force_flush_buffered_events()` and flushes all input. So their code should still work as before, but with no agile input.

If writing a mainloop from c++, then they have the choice of using legacy mode and calling `flush_buffered_events()` (as before), or calling the new functions directly, following the example in `Main::iteration()`, which would support agile input. For them to do this, they would need to run `main_timer_sync`, or do the calculations themselves, in order to get the physics tick logical times.

In a way I'm suspecting we will need some feedback from these special case projects in order to know how they want to best handle this. This is part of the reason why I was initially exposing `force_flush_buffered_events()`, because although it is discouraged, it may be useful for people writing their own mainloop who want to take the old style approach. But maybe just using `flush_buffered_events()` and legacy mode caters for this scenario. :thinking:

Reply:
I decided to reinstate the old functionality for `flush_buffered_events()` and not call it on Android in the mainloop. This is backward compatible, but requires care to not call it regularly on platforms that desire to support agile input.